src/PostgREST/Request/Preferences.hs

```diff
@@ -1,16 +1,69 @@
-module PostgREST.Request.Preferences where
+module PostgREST.Request.Preferences
+  ( Preferences(..)
+  , PreferCount(..)
+  , PreferParameters(..)
+  , PreferRepresentation(..)
+  , PreferResolution(..)
+  , PreferTransaction(..)
+  , fromHeaders
+  , ToAppliedHeader(..)
+  ) where
+
+import qualified Data.ByteString           as BS
+import qualified Data.Map                  as Map
+import qualified Network.HTTP.Types.Header as HTTP
 
-import GHC.Show
 import Protolude
 
 
+data Preferences
+  = Preferences
+    { preferResolution     :: Maybe PreferResolution
+    , preferRepresentation :: PreferRepresentation
+    , preferParameters     :: Maybe PreferParameters
+    , preferCount          :: Maybe PreferCount
+    , preferTransaction    :: Maybe PreferTransaction
```

Comment:
I don't understand why we're getting a codecov warning here that those 5 lines are not covered. Are those getters not what is used in `ApiRequest`?

Reply:
I think that's the `RecordWildCards` bug we already struggled with

Reply:
Right, I remember... gotta get used to things again :D